### PR TITLE
feat: add indexAutomatically option to disable middleware hooks

### LINKED ISSUE #49: Disabling automatic indexing of documents
I wonder if it is possible to disable the automatic indexing on CRUD operations and only have on-demand-indexing.

We use mongodb change streams to do the indexing, but I would still like to use this lib for some of teh convenience methods it offers.

### PR DESCRIPTION
**Pull request by florianbepunkt**
closes #49 

indexAutomatically option is set to true by default. when set to false, middleware hooks are not fired which allows for more fine grained control over when a document should be indexed/removed

### CODE REVIEW

**Comment by nodkz** [comment]
Please rename option to `onlyOnDemandIndexing` and add info about it to README.

Thanks.

File: README.md [hunk 1]
```diff
@@ -64,35 +64,35 @@ The examples below use the version 5 syntax.
...
+- `idsOnly` - whether or not returning only mongo ids in `esSearch`.
+- `countOnly` - whether or not returning only the count value in `esCount`.
+- `mappingSettings` - default settings to use with `esCreateMapping`.
+- `refreshDelay` - time in ms to wait after `esRefresh`. Defaults to 0.
+- `script` - whether or not the inline script are enabled in elasticsearch. Defaults to false.
+- `bulk` - options to use when synchronising.
+- `bulk.batch` - [batchSize](https://docs.mongodb.com/manual/reference/method/cursor.batchSize/) to use on synchronise options. Defaults to 50.
+- `bulk.size` - bulk element count to wait before calling `client.bulk` function. Defaults to 1000.
+- `bulk.delay` - idle time to wait before calling the `client.bulk` function. Defaults to 1000.
+- `onlyOnDemandIndexing` - whether or not to automatically index on CRUD operations. If set to false mexp middleware hooks for save, delete, update do not fire. Defaults to true.
```
Comment (nodkz) [inline README.md]:
```diff
- whether or not to automatically index on CRUD operations. If set to false mexp middleware hooks for save, delete, update do not fire. Defaults to true.
+ whether or not to automatically index on CRUD operations. If set to true mexp middleware hooks for save, delete, update do not fire. Defaults to false.
```

`onlyOnDemandIndexing: true` - means that indexing should be called manually. Current implementation breaks my mind ;)) Please fix it.

File: lib/index.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ module.exports = function(schema, options, version) {
   schema.methods.esUnset = unsetFields;
   schema.methods.esRemove = removeDoc;
 
-  if (options.indexAutomatically) {
+  if (options.onlyOnDemandIndexing) {
```
Comment (nodkz) [inline lib/index.js]: Should be:
```js
if (!options.onlyOnDemandIndexing) {
```

File: lib/index.js [hunk 3]
```diff
@@ -10,8 +10,8 @@ module.exports = function(schema, options, version) {
   // clone main level of options (does not clone deeper)
   options = utils.highClone(options);
 
-  if (!Object.prototype.hasOwnProperty.call(options, 'indexAutomatically')) {
-    options.indexAutomatically = true;
+  if (!Object.prototype.hasOwnProperty.call(options, 'onlyOnDemandIndexing')) {
```
Comment (nodkz) [inline lib/index.js]: This block of code can be removed entirely. See comment below.

File: lib/index.js (hunk 2 above)
Comment (nodkz) [reply lib/index.js]: Dont forget to change README
```diff
- whether or not to automatically index on CRUD operations. If set to false mexp middleware hooks for save, delete, update do not fire. Defaults to true.
+ whether or not to automatically index on CRUD operations. If set to true mexp middleware hooks for save, delete, update do not fire. Defaults to false.
```

**Comment by nodkz** [comment]
FYI You also may use Filtered Indexing https://github.com/jbdemonte/mongoose-elasticsearch-xp#filtered-indexing. When some field in model has proper values, then doc be send or removed from ES index.

**Comment by nodkz** [comment]
Make proper changes myself directly to master

**Comment by nodkz** [comment]
@florianbepunkt thanks for PR

**Comment by nodkz** [comment]
@florianbepunkt  Version 5.6.0 was just published to npm.
Please try it with your app.

**Comment by florianbepunkt** [comment]
Will do so this week. Thank you.

**Comment by florianbepunkt** [comment]
@nodkz Working great, thank you

**Pull request merged**